sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -84,19 +87,21 @@ export class IER12ProcessingService {
     }
     const uploadResult: IER12UploadResult[] = [];
     try {
-      this.logger.log("Creating IER 12 content...");
+      processSummary.info("Creating IER 12 content.");
       for (const [institutionCode, ierRecords] of Object.entries(fileRecords)) {
+        const uploadProcessSummary = new ProcessSummary();
+        processSummary.children(uploadProcessSummary);
         const ierUploadResult = await this.uploadIER12Content(
           institutionCode,
           ierRecords,
+          uploadProcessSummary,
         );
         uploadResult.push(ierUploadResult);
       }
-    } catch (error) {
-      this.logger.error(`Error while uploading content for IER 12: ${error}`);
-      // TODO: On error, the error message must added to the upload result and
-      // processing must continue for the next institution without aborting.
-      throw error;
+    } catch (error: unknown) {
+      const errorMessage = "Error while uploading content for IER 12.";
+      this.logger.error(errorMessage, error);
+      processSummary.error(errorMessage, error);
```

Comment:
What could be the benefit of not throwing the error here?

Reply:
First, the previous log that stated "processing must continue for the next institution without aborting." was misleading because this block is the last one before wrapping up the operation.
Throwing or not throwing here will not change the fact that the job will retry, but not throwing will allow the `IER12UploadResult` to be returned and logged before the job fails.